PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,3 @@
+(Strongly consider adding notes for this change)
```

Comment:
Maybe point to a document explaining what a note is and/or we should encourage ppl to document each commits and squash them appropriately.